src/common/models/utils.ts

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash';
+
+// TODO(michael): We should perhaps create a Timeseries class that wraps an
+// Array<Date> and Array<number | null> and hang these utility functions off of
+// that.
+
+export function indexOfLastValue<T>(data: Array<T | null>): number | null {
+  for (let i = data.length - 1; i >= 0; i--) {
+    if (data[i] !== null && data[i] !== undefined) {
+      return i;
+    }
+  }
+  return null;
+}
+
+export function lastValue<T>(data: Array<T | null>): T | null {
+  const i = indexOfLastValue(data);
+  return i === null ? null : data[i]!;
+}
+
+/**
+ * Replaces all values after the specified date with nulls.
+ */
+export function omitDataAfterDate<T>(
```

Comment:
Note: The diff is hidden since I moved this function, but I changed it to omitDataAfterDate() instead of omitDataAtOrAfterDate() since I noticed the same thing you did that it was eating the latest data point.